framework/Samplers/MonteCarlo.py

```diff
@@ -130,25 +130,24 @@ def localGenerateInput(self,model,myInput):
       reducedDim = self.variables2distributionsMapping[key]['reducedDim']
       weight = 1.0
       if totDim == 1:
-        for var in self.distributions2variablesMapping[dist]:
-          varID  = utils.first(var.keys())
-          if self.samplingType == 'uniform':
-            distData = self.distDict[key].getCrowDistDict()
-            if ('xMin' not in distData.keys()) or ('xMax' not in distData.keys()):
-              self.raiseAnError(IOError,"In the Monte-Carlo sampler a uniform sampling type has been chosen; however, one or more distributions have not specified either the lowerBound or the upperBound")
-            lower = distData['xMin']
-            upper = distData['xMax']
-            rvsnum = lower + (upper - lower) * randomUtils.random()
-            epsilon = (upper-lower)/self.limit
-            midPlusCDF  = self.distDict[key].cdf(rvsnum + epsilon)
-            midMinusCDF = self.distDict[key].cdf(rvsnum - epsilon)
-            weight *= midPlusCDF - midMinusCDF
-          else:
-            rvsnum = self.distDict[key].rvs()
-          self.inputInfo['SampledVarsPb'][key] = self.distDict[key].pdf(rvsnum)
-          for kkey in varID.strip().split(','):
-            self.values[kkey] = np.atleast_1d(rvsnum)[0]
-          self.inputInfo['ProbabilityWeight-' + varID] = 1.
+        if self.samplingType == 'uniform':
+          distData = self.distDict[key].getCrowDistDict()
+          if ('xMin' not in distData.keys()) or ('xMax' not in distData.keys()):
+            self.raiseAnError(IOError,"In the Monte-Carlo sampler a uniform sampling type has been chosen;"
```

Comment:
is this possible? Each distribution sets its own lowerBound and upperBound as support. Right?
In addition, are they still call xMin,xMax? I thought we convert everything to be lowerBOund and upperBound

Reply:
For my understanding, this is only used in uniform sampling. In this case, we need to require the distribution provide  lowerBound and upperBound, otherwise, we can not use uniform sampling. Regarding ```xMin and xMax```, they are just mapping between lowerBound and upperBound, and they are only used for internal, not exposed to the user. And I agree with you that I also think we have already convert everything to be lowerBound and upperBound. Please correct me if I was wrong @mandd